docs/references/architecture/adr-113-modular-transaction-hashing.md

```diff
@@ -127,17 +127,34 @@ can only do so once before launching their app. If they attempt to upgrade
 after without a hard fork, the resulting hashes won't match. A hard fork would
 work.
 
-The hashing function needs to be added to `Header`, so that light clients are aware of
-the function to use for verification.
+### IBC
+
+In IBC, a zone is running a light client for each zone it's connected to. If we
+allow changing the hash, light clients need to be aware of that.
+
+Specifically, the `light.Client` constructor needs to be updated:
 
 ```go
-type Header struct {
-    // ...
-	Hash crypto.Hash `json:"hash"`
-}
+func NewClient(
+	ctx context.Context,
+	chainID string,
+	trustOptions TrustOptions,
+	primary provider.Provider,
+	witnesses []provider.Provider,
+	trustedStore store.Store,
+    hash crypto.Hash,
+	options ...Option,
+) (*Client, error) {
 ```
 
-Light clients would then use ^ when calculating validators hash or header's hash.
+To help discover the hash function used by the other zone, we can add a new
+field to the genesis file:
+
+```json
+{
+    "hash": 5, // crypto.SHA256 (https://pkg.go.dev/crypto#Hash)
```

Comment:
What does "5" mean here?

Reply:
Go's stdlib `Hash` https://pkg.go.dev/crypto#Hash. We can also use strings like `SHA256` if that's more practical